pkg/cluster/util.go

```diff
@@ -271,6 +271,33 @@ func (c *Cluster) getTeamMembers(teamID string) ([]string, error) {
 	return members, nil
 }
 
+// Returns annotations to be passed to child objects
+func (c *Cluster) annotationsSet(annotations map[string]string) map[string]string {
+
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	pgCRDAnnotations := c.ObjectMeta.Annotations
+
+	// allow to inherit certain labels from the 'postgres' object
+	if pgCRDAnnotations != nil {
+		for k, v := range pgCRDAnnotations {
+			for _, match := range c.OpConfig.InheritedAnnotations {
+				if k == match {
+					annotations[k] = v
+				}
+			}
+		}
+	}
+
+	if len(annotations) > 0 {
+		return annotations
+	}
+
+	return nil
+}
+
```

Comment:
Not sure if returning nil is used anywhere, can simply return annotations.

Reply:
it should return nil if there are no Annotations defined in the config and on the resource. For example `TestGeneratePodDisruptionBudget` fails for me if I return an empty map as this test does not define any Annotations at all. I have also added this behavior to the `AnnotationsToPropagate` function, too.